File: internal/dto/course/question_bank.go
package course

// QuestionBankItem 课程题库题目实体（教师可手工维护，也可由 AI 解析文件后生成）。
type QuestionBankItem struct {
	ID              string `json:"id"`
	CourseID        string `json:"course_id"`
	QuestionType    string `json:"question_type"`
	Stem            string `json:"stem"`
	ReferenceAnswer string `json:"reference_answer"`
	SourceFileName  string `json:"source_file_name,omitempty"`
	CreatedAt       string `json:"created_at,omitempty"`
	UpdatedAt       string `json:"updated_at,omitempty"`
}

// ListQuestionBankItemsResp 题库分页/列表响应（page、page_size、total 与 items 一并返回便于前端分页）。
type ListQuestionBankItemsResp struct {
	Items    []QuestionBankItem `json:"items"`
	Total    int64              `json:"total"`
	Page     int                `json:"page"`
	PageSize int                `json:"page_size"`
}

// ParseQuestionBankImportResp AI 解析上传文本后的题目草稿，不入库，供教师编辑后再确认。
type ParseQuestionBankImportResp struct {
	Questions []CreateQuestionBankItemReq `json:"questions"`
}

// ConfirmQuestionBankImportReq 教师确认后的批量入库请求。
type ConfirmQuestionBankImportReq struct {
	SourceFileName string                      `json:"source_file_name"`
	Questions      []CreateQuestionBankItemReq `json:"questions" binding:"required"`
}

// ConfirmQuestionBankImportResp 批量确认入库结果。
type ConfirmQuestionBankImportResp struct {
	CreatedCount int               `json:"created_count"`
	Items        []QuestionBankItem `json:"items"`
}

// CreateQuestionBankItemReq 手工新增题目请求。
type CreateQuestionBankItemReq struct {
	QuestionType    string `json:"question_type" binding:"required"`
	Stem            string `json:"stem" binding:"required"`
	ReferenceAnswer string `json:"reference_answer" binding:"required"`
}

// CreateQuestionBankItemResp 新增题目响应。
type CreateQuestionBankItemResp struct {
	ID string `json:"id"`
}

// UpdateQuestionBankItemReq 更新题目请求。
type UpdateQuestionBankItemReq struct {
	QuestionType    string `json:"question_type" binding:"required"`
	Stem            string `json:"stem" binding:"required"`
	ReferenceAnswer string `json:"reference_answer" binding:"required"`
}

File: frontend/src/services/questionBank.ts
import { request, type ApiEnvelope } from './api'

export type QuestionBankItem = {
  id: string
  course_id: string
  question_type: string
  stem: string
  reference_answer: string
  source_file_name?: string
  created_at?: string
  updated_at?: string
}

export type QuestionBankListResp = {
  items: QuestionBankItem[]
  total: number
  page: number
  page_size: number
}

export type CreateQuestionPayload = {
  question_type: string
  stem: string
  reference_answer: string
}

export type QuestionBankListQuery = {
  page?: number
  page_size?: number
  keyword?: string
  question_type?: string
}

const API_BASE_URL = import.meta.env.VITE_API_BASE_URL ?? '/api/v1'

export async function listQuestionBank(token: string, courseId: string, q: QuestionBankListQuery = {}) {
  const sp = new URLSearchParams()
  if (q.page != null && q.page > 0) sp.set('page', String(q.page))
  if (q.page_size != null && q.page_size > 0) sp.set('page_size', String(q.page_size))
  if (q.keyword?.trim()) sp.set('keyword', q.keyword.trim())
  if (q.question_type?.trim()) sp.set('question_type', q.question_type.trim())
  const qs = sp.toString()
  const path = `/teacher/courses/${courseId}/question-bank${qs ? `?${qs}` : ''}`
  return request<QuestionBankListResp>(path, { method: 'GET', token })
}

export async function createQuestion(token: string, courseId: string, payload: CreateQuestionPayload) {
  return request<{ id: string }>(`/teacher/courses/${courseId}/question-bank`, {
    method: 'POST',
    token,
    body: payload,
  })
}

export async function updateQuestion(token: string, itemId: string, payload: CreateQuestionPayload) {
  return request<unknown>(`/teacher/question-bank/items/${itemId}`, {
    method: 'PUT',
    token,
    body: payload,
  })
}

export async function deleteQuestion(token: string, itemId: string) {
  return request<unknown>(`/teacher/question-bank/items/${itemId}`, {
    method: 'DELETE',
    token,
  })
}

type ParseImportData = { questions: CreateQuestionPayload[] }

/** 上传文本并由模型解析为草稿（不入库）。 */
export async function parseImportFile(token: string, courseId: string, file: File, qaModelId?: string) {
  const form = new FormData()
  form.append('file', file)
  if (qaModelId) form.append('qa_model_id', qaModelId)
  const resp = await fetch(`${API_BASE_URL}/teacher/courses/${courseId}/question-bank/import/parse`, {
    method: 'POST',
    headers: token ? { Authorization: `Bearer ${token}` } : undefined,
    body: form,
  })
  const json = (await resp.json()) as ApiEnvelope<ParseImportData>
  if (!resp.ok) {
    throw new Error(json?.message || `解析失败（${resp.status}）`)
  }
  return (json?.data?.questions ?? []) as CreateQuestionPayload[]
}

/** 确认后将题目批量写入题库。 */
export async function confirmImportBatch(
  token: string,
  courseId: string,
  body: { source_file_name?: string; questions: CreateQuestionPayload[] },
) {
  return request<{ created_count: number }>(`/teacher/courses/${courseId}/question-bank/import/confirm`, {
    method: 'POST',
    token,
    body: {
      source_file_name: body.source_file_name ?? '',
      questions: body.questions,
    },
  })
}
